tests/kernel/mem_pool/sys_mem_pool/src/main.c

```diff
@@ -21,9 +21,13 @@
 
 
 K_MUTEX_DEFINE(pool_mutex);
+#ifdef CONFIG_APP_SHARED_MEM
+SYS_MEM_POOL_DEFINE(pool, &pool_mutex, BLK_SIZE_MIN, BLK_SIZE_MAX,
+		    BLK_NUM_MAX, BLK_ALIGN, data_smem_ztest_part0_data);
```

Comment:
There's got to be a better way to do this, maybe a helper macro for the section name?

Reply:
Nope not possible. the macro SYS_MEM_POOL_DEFINE will take the last arguement as a sting. 
if we add a macro gcc will not preprocess and add it.